src/components/MenuBars/MenuBars.tsx

```diff
@@ -54,7 +55,7 @@ function MenuBars() {
         {"menu-bars--bottom": document.getElementById("menu-bars")?.classList.contains("menu-bars--bottom")}
       )}
     >
-      <section className="menu user-menu">
+      <section className={classNames("menu", "user-menu", {"menu-animation": animate})} onTransitionEnd={() => setAnimate(false)}>
```

Comment:
Since onTransitionEnd will be called upon each CSS property `setAnimate(false)` will be called after `0.4s`. You can check for the CSS property name here and read it from the `event` which will be passed to the inner function before calling `setAnimate` or use the React animations addon here https://reactjs.org/docs/animation.html

Seems like a bit of overhead but would be the proper solution here I guess

Reply:
... or you can try to unify the transition durations

![Head explodes](https://media.giphy.com/media/kH66pmpo7NjIgtI696/giphy.gif)

Reply:
Good point. Thank you. Didn't thought about `event bubbling`. I will consider all the suggestions and choose the most promising one 😄

Reply:
I decided to filter the events based on their `classNames`. But we can keep the second approach (`ReactTransitionGroup`) in mind and replace the current solution if needed in the future.